astropy/io/ascii/tests/test_ecsv.py

```diff
@@ -678,6 +701,25 @@ def test_read_bad_datatype():
         Table.read(txt, format='ascii.ecsv')
 
 
+def test_read_bad_datatype_v09():
+    """Test a malformed ECSV file for V0.9"""
+    txt = """\
+# %ECSV 0.9
+# ---
+# datatype:
+# - {name: a, datatype: object}
+# schema: astropy-2.0
+a
+fail
+[3,4]"""
+    try:
+        Table.read(txt, format='ascii.ecsv')
+    except ValueError:
+        pytest.fail("V0.9 ECSV files should not raise ValueError"
```

Comment:
What if it is another kind of error? Is it supposed to pass then?

Reply:
It's not removed from the test, so resolving this.